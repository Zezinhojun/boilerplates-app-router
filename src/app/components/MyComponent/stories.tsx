import { Meta, StoryObj } from "@storybook/nextjs";
import MyComponent from ".";

export default {
  title: "MyComponent",
  component: MyComponent,
} as Meta;

export const Default: StoryObj = {};
