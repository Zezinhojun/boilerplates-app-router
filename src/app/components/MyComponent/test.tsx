import { render, screen } from "@testing-library/react";

import MyComponent from ".";

describe("<MyComponent/>", () => {
  it("Should render the heading", () => {
    const { container } = render(<MyComponent />);

    expect(
      screen.getByRole("heading", { name: /MyComponent/i }),
    ).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
