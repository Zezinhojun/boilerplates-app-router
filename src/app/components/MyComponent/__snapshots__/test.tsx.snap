// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`<MyComponent/> Should render the heading 1`] = `
<main
  class=""
>
  <h1>
    MyComponent
  </h1>
</main>
`;
